File: content/snippets/cpp/interview_lcci_01_03_string_to_url.cpp
#include <iostream>
#include <string>

std::string urlify(const std::string& s, int trueLength) {
    std::string out;
    out.reserve(static_cast<size_t>(trueLength) * 3);

    for (int i = 0; i < trueLength; ++i) {
        if (s[static_cast<size_t>(i)] == ' ') {
            out += "%20";
        } else {
            out.push_back(s[static_cast<size_t>(i)]);
        }
    }

    return out;
}

int main() {
    const std::string input = "Mr John Smith    ";
    std::cout << urlify(input, 13) << '\n';
    return 0;
}

File: content/snippets/scala3/interview_lcci_01_03_string_to_url.scala
object Lcci0103StringToUrl:
  def urlify(s: String, trueLength: Int): String =
    val b = new StringBuilder
    for (i <- 0 until trueLength) {
      if (s(i) == ' ') {
        b.append("%20")
      } else {
        b.append(s(i))
      }
    }
    b.toString()

  @main def main(): Unit =
    println(urlify("Mr John Smith    ", 13))
